src/maidsafe/vault/mpid_manager/service.h

```diff
@@ -127,14 +133,25 @@ class MpidManagerService {
   void HandleSyncedRemoveAccount(
            std::unique_ptr<MpidManager::UnresolvedRemoveAccount>&& synced_action);
 
+// =============== Account Transfer ==============================================================
+  void TransferAccount(const NodeId& destination,
+                       const std::vector<MpidManager::KVPair>& accounts);
+  void HandleAccountTransferEntry(const std::string& serialised_account,
+                                  const routing::SingleSource& sender);
+  void HandleAccountTransfer(const MpidManager::KVPair& account);
+  void HandleAccountQuery(const ImmutableData::Name& name,
+                          const NodeId& sender,
+                          const NodeId& receiver);
+
   routing::Routing& routing_;
-  mutable std::mutex accumulator_mutex_, nodes_change_mutex_, pending_account_mutex_;
+  mutable std::mutex accumulator_mutex_, nodes_change_mutex_, pending_account_mutex_, mutex_;
```

Comment:
supposed to protect the global status -- stopped, and other global data.
However, I doubt whether it is a good idea to have an individual mutex to protect only one global data.
We had accumulator_mutex_ because it is supposed to be passed down, and then use mutex_ to protect persona service global member data.
Using individual mutex (such as nodes_change_mutex_ and pending_account_mutex_) may improve the performance but will also introduce additional risk of dead-lock.

Personally, I prefer a simpler model : individual mutex if required to be passed down/out, then one mutex for all internal protection.